src/collar/oc_bookmarks.lsl

```diff
@@ -1,523 +1,606 @@
-//integer CMD_BLOCKED               = 520;
-
-integer NOTIFY                     = 1002;
-//integer SAY                        = 1004;
-integer REBOOT                     = -1000;
-integer LINK_DIALOG                = 3;
-integer LINK_RLV                   = 4;
-integer LINK_SAVE                  = 5;
-integer LINK_UPDATE                = -10;
-integer LM_SETTING_SAVE            = 2000;
-integer LM_SETTING_RESPONSE        = 2002;
-integer LM_SETTING_DELETE          = 2003;
-integer MENUNAME_REQUEST           = 3000;
-integer MENUNAME_RESPONSE          = 3001;
-integer MENUNAME_REMOVE            = 3003;
-integer RLV_CMD                    = 6000;
-integer RLV_OFF                    = 6100;
-integer RLV_ON                     = 6101;
-integer DIALOG                     = -9000;
-integer DIALOG_RESPONSE            = -9001;
-integer DIALOG_TIMEOUT             = -9002;
+string UPMENU = "BACK";
+key g_kCommander;
+
+list PLUGIN_BUTTONS = ["SAVE", "PRINT", "REMOVE"];
+
+// MESSAGE MAP
+//integer CMD_ZERO = 0;
+integer CMD_OWNER = 500;
+//integer CMD_TRUSTED = 501;
+integer CMD_GROUP = 502;
+integer CMD_WEARER = 503;
+//integer CMD_EVERYONE = 504;
+//integer CMD_RLV_RELAY = 507;
+//integer CMD_SAFEWORD = 510;
+//integer CMD_BLOCKED = 520;
+
+integer NOTIFY = 1002;
+//integer SAY = 1004;
+integer REBOOT = -1000;
+
+integer LINK_DIALOG = 3;
+integer LINK_RLV = 4;
+integer LINK_SAVE = 5;
+integer LINK_UPDATE = -10;
+
+integer LM_SETTING_SAVE = 2000;
+integer LM_SETTING_RESPONSE = 2002;
+integer LM_SETTING_DELETE = 2003;
+
+integer MENUNAME_REQUEST = 3000;
+integer MENUNAME_RESPONSE = 3001;
+integer MENUNAME_REMOVE = 3003;
+
+integer RLV_CMD = 6000;
+integer RLV_OFF = 6100;
+integer RLV_ON = 6101;
+
+integer DIALOG = -9000;
+integer DIALOG_RESPONSE = -9001;
+integer DIALOG_TIMEOUT = -9002;
+
+/*
+integer g_bProfiled;
+Debug(string sStr) {
+  // If you delete the first // from the preceeding and following  lines,
+  // profiling is off, debug is off, and the compiler will remind you to
+  // remove the debug calls from the code, we're back to production mode
+  if (!g_bProfiled) {
+    g_bProfiled = TRUE;
+    llScriptProfiler(TRUE);
+  }
+  llOwnerSay(llGetScriptName() + "(min free:" + (string)(llGetMemoryLimit() - llGetSPMaxMemory()) + ")[" + (string)llGetFreeMemory() + "] :\n" + sStr);
+}
+*/
 
 Dialog(key kRCPT, string sPrompt, list lChoices, list lUtilityButtons, integer iPage, integer iAuth, string sMenuType) {
-    key kMenuID = llGenerateKey();
-    llMessageLinked(LINK_DIALOG, DIALOG, (string)kRCPT + "|" + sPrompt + "|" + (string)iPage + "|" + llDumpList2String(lChoices, "`") + "|" + llDumpList2String(lUtilityButtons, "`") + "|" + (string)iAuth, kMenuID);
-    integer iIndex = llListFindList(g_lMenuIDs, [kRCPT]);
-    if (~iIndex) g_lMenuIDs = llListReplaceList(g_lMenuIDs, [kRCPT, kMenuID, sMenuType], iIndex, iIndex + g_iMenuStride - 1);
-    else g_lMenuIDs += [kRCPT, kMenuID, sMenuType];
+  key kMenuID = llGenerateKey();
```

Comment:
It appears you used a search and replace for  a certain number of spaces. LSL infact does not use tabs for indentation, it uses five spaces.